quint/src/names/collector.ts

```diff
@@ -154,6 +154,11 @@ export class NameCollector implements IRVisitor {
       return
     }
 
+    if (def.qualifiedName) {
+      const newTable = new Map([...moduleTable.entries()])
```

Comment:
I believe you that it fixes the issue, but I don't understand why it fixes the issue. Could you please add a comment here on how `export Foo.*` and `import ...` are cooperating here.